tests/unit/aiplatform/test_pipeline_jobs.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import json
+
+from unittest import mock
+from importlib import reload
+from unittest.mock import patch
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud import aiplatform
+from google.cloud import storage
+
+from google.cloud.aiplatform import pipeline_jobs
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform.utils import json_utils
+
+from google.cloud.aiplatform_v1beta1.services.pipeline_service import (
+    client as pipeline_service_client_v1beta1,
+)
+
+from google.cloud.aiplatform_v1beta1.types import (
+    pipeline_job as gca_pipeline_job_v1beta1,
+    pipeline_state as gca_pipeline_state_v1beta1,
+)
+
+_TEST_PROJECT = "test-project"
+_TEST_LOCATION = "us-central1"
+_TEST_PIPELINE_JOB_ID = "sample-test-pipeline-202111111"
+_TEST_GCS_BUCKET_NAME = "my-bucket"
+_TEST_CREDENTIALS = mock.Mock(spec=auth_credentials.AnonymousCredentials())
+_TEST_SERVICE_ACCOUNT = "abcde@my-project.iam.gserviceaccount.com"
+
+_TEST_JOB_SPEC_PATH = f"gs://{_TEST_GCS_BUCKET_NAME}/job_spec.json"
+_TEST_PARENT = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}"
+_TEST_NETWORK = f"projects/{_TEST_PROJECT}/global/networks/{_TEST_PIPELINE_JOB_ID}"
+
+_TEST_PIPELINE_JOB_NAME = f"projects/{_TEST_PROJECT}/locations/{_TEST_LOCATION}/pipelineJobs/{_TEST_PIPELINE_JOB_ID}"
+
+_TEST_MACHINE_TYPE = "n1-standard-4"
+_TEST_ACCELERATOR_TYPE = "NVIDIA_TESLA_P100"
+_TEST_ACCELERATOR_COUNT = 2
+_TEST_STARTING_REPLICA_COUNT = 2
+_TEST_MAX_REPLICA_COUNT = 12
+
+_TEST_PIPELINE_PARAMETER_VALUES = {"name_param": {"stringValue": "hello"}}
```

Comment:
Are we expecting user to pass this? or just `{"name_param":  "hello"}` (the KFP client behavior)?

Reply:
Updated, thanks!